sympy/integrals/integrals.py

```diff
@@ -516,6 +517,28 @@ def try_meijerg(function, xab):
                             function = ret
                             continue
 
+            if not isinstance(antideriv,Integral) and antideriv is not None:
+                sym = xab[0]
+                if antideriv.is_Add:
+                    terms = antideriv.args
+                else:
+                    terms = tuple((antideriv,))
+                for term in terms:
+                    logset = term.atoms(log)
+                    if len(logset) != 0:
+                        if logset.pop().args[0].has(tan):
+                            break
+                    else:
+                        argtan = []
+                        for part in term.atoms(tan) | term.atoms(cot):
```

Comment:
`tan` and `cot` may appear in an antiderivative with no `atan` (or `log`). There is no reason to add a floor term to such expressions.
I would  start by searching `atan` nodes and then take only those whose argument is a linear function of `tan` or `cot`. There would be no need to explicitly search and remove `log` (or any other) terms.

Reply:
yup!

Reply:
well even if `tan` or `cot` will be present in integral then if it is continuous then `P` will be zero.

Reply:
@jksuom I need some help in how to check for linear funtion of `tan` or `cot` as argument can be of type: a*tan(x), a*tan(x)+b, sqrt(c)(a*tan(x)+sqrt(b))/c, etc.